backend/app/services/manifest/Neo4jManifest.scala

```diff
@@ -1057,9 +1057,11 @@ class Neo4jManifest(driver: Driver, executionContext: ExecutionContext, queryLog
       val relationshipsDeleted = counters.relationshipsDeleted()
       val propertiesSet = counters.propertiesSet()
 
-      if (relationshipsDeleted  < 1 || propertiesSet < 1 ) {
+      // if the blob has an EXTRACTION_FAILURE relation then that will be deleted and we expect to set at least one property
+      // (1 property if there is a TODO, more if there's a PROCESSING_EXTERNALLY relation to replace with a TODO)
+      if (relationshipsDeleted != 0 && propertiesSet < 1 ) {
```

Comment:
In the case where we're reprocessing a file that doesn't have an EXTRACTION_FAILURE, do we still want to check that propertiesSet > 0? If so, maybe this condition could just be `if (propertiesSet == 0)`?

Reply:
In that case, we don't want to check propertiesSet at all as we don't expect any changes from this function.

I think that this is confusing as one function is essentially doing two different things depending on what's happening in the database. I'll add a ticket to split this up into something easier to follow